gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -592,6 +593,41 @@ ChannelCredentials createS2ASecuredChannelCredentials() {
     }
   }
 
+  boolean isMtlsS2AHardBoundTokensEnabled() {
+    if (!useS2A) {
+      // If S2A cannot be used, {@code HardBoundTokenTypes.MTLS_S2A} hard bound tokens should not be
+      // used
+      return false;
+    }
+    if (allowedHardBoundTokenTypes == null
+        || credentials == null
+        || !(credentials instanceof ComputeEngineCredentials)) {
+      return false;
+    }
+    for (HardBoundTokenTypes boundTokenTypes : allowedHardBoundTokenTypes) {
+      if (boundTokenTypes == HardBoundTokenTypes.MTLS_S2A) {
+        return true;
+      }
+    }
+    return false;
```

Comment:
Can we use [Stream.anyMatch](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#anyMatch-java.util.function.Predicate-) here?

Reply:
Yes, we can, done in https://github.com/googleapis/sdk-platform-java/pull/3591/commits/7c20422a57882b0243b298ef8e6f33293e4aa61a